### ISSUE TITLE
CeylonFile.waitForUpToDatePhasedUnit() is super slow

### ISSUE THREAD

**Issue by bjansen** [ceylon/ceylon-ide-intellij#442]
While testing navigation from Java calls to Ceylon source declarations, the UI sometimes freezes because `CeylonFile.waitForUpToDatePhasedUnit()` takes an incredible amount of time to return (way more than the 4s timeout configured in `CeylonSourceNavigator`).

For example, in `org.intellij.plugins.ceylon.ide.action.CeylonAddingFilesAction`, when I ctrl-click on `projects.getProject()` in line 59, the UI hangs for more than 20s, then navigates to the decompiled class, meaning we couldn't retrieve a phased unit for `CeylonProjects.ceylon`.

[threadDump-20160831-120352.txt](https://github.com/ceylon/ceylon-ide-intellij/files/446925/threadDump-20160831-120352.txt)
[threadDump-20160831-120357.txt](https://github.com/ceylon/ceylon-ide-intellij/files/446926/threadDump-20160831-120357.txt)
[threadDump-20160831-120402.txt](https://github.com/ceylon/ceylon-ide-intellij/files/446927/threadDump-20160831-120402.txt)

If I manually open `CeylonProjects.ceylon`, then try to ctrl-click on `projects.getProject()` again, it navigates instantly to the source declaration.

**Comment by davidfestal** [comment]
I cannot reproduce the problem, even when a model update is running, or even just after opening the project.

**Comment by davidfestal** [comment]
@bjansen  do you confirm that we can close this one ?

**Comment by bjansen** [comment]
In IdeaCeylonParser, I see this code:

```
root.putUserData(parserConstants._POST_PARSE_ACTION, postParseAction);
return root;
```

We expect that post-parse action to be called later, to do `future.set(externalPu);` but in fact it's never called when I test the scenario described above.

It looks like IDEA never calls `CeylonParserDefinition.createElement()`, so we don't retrieve+call the post-parse action.

**Comment by bjansen** [comment]
Fixed by e96a32ad5e42d2a6696f3981f5a493f39ffa0faa.
